examples/finnegans_wake_demo/finnegans-wake-demo-lynx.py

```diff
@@ -0,0 +1,191 @@
+)
+
+# Here are the Policy details.
+policy_end_datetime = maya.now() + datetime.timedelta(days=5)
+m, n = 2, 3
+label = b"secret/files/and/stuff"
+
+# Alice can get the public key even before creating the policy.
+# From this moment on, any Data Source that knows the public key
+# can encrypt data originally intended for Alice, but that can be shared with
+# any Bob that Alice grants access.
+policy_pubkey = ALICE.get_policy_encrypting_key_from_label(label)
+
+# Alice already knows Bob's public keys from a side-channel.
+bob = Bob.from_public_keys(encrypting_key=encrypting_key, verifying_key=verifying_key)
+
+# Alice grants access to bob.
+policy = ALICE.grant(
+    bob=bob,
+    label=label,
+    m=m,
+    n=n,
+    rate=Web3.toWei(50, 'gwei'),
+    expiration=policy_end_datetime
+)
+
+# Alice puts her public key somewhere for Bob to find later...
+alices_pubkey_bytes_saved_for_posterity = bytes(ALICE.stamp)
+
+# ...and then disappears from the internet.
+ALICE.disenchant()
+del ALICE
+del bob
+
+#####################
+# some time passes. #
+# ...               #
+#                   #
+# ...               #
+# And now for Bob.  #
+#####################
+
+BOB.join_policy(label, alices_pubkey_bytes_saved_for_posterity)
+
+# Now that Bob has joined the Policy, let's show how Enrico the Encryptor
+# can share data with the members of this Policy and then how Bob retrieves it.
+# In order to avoid re-encrypting the entire book in this demo, we only read some lines.
+with open(BOOK_PATH, 'rb') as file:
+    finnegans_wake = file.readlines()[:NUMBER_OF_LINES_TO_REENCRYPT]
+
+print()
+print("**************James Joyce's Finnegan's Wake**************")
+print()
+print("---------------------------------------------------------")
+
+for counter, plaintext in enumerate(finnegans_wake):
+
+    #########################
+    # Enrico, the Encryptor #
+    #########################
+    enrico = Enrico(policy_encrypting_key=policy_pubkey)
+
+    # In this case, the plaintext is a
+    # single passage from James Joyce's Finnegan's Wake.
+    # The matter of whether encryption makes the passage more or less readable
+    # is left to the reader to determine.
+    single_passage_ciphertext, _signature = enrico.encrypt_message(plaintext)
+    data_source_public_key = bytes(enrico.stamp)
+    del enrico
+
+    ###############
+    # Back to Bob #
+    ###############
+
+    enrico_as_understood_by_bob = Enrico.from_public_keys(
+        verifying_key=data_source_public_key,
+        policy_encrypting_key=policy_pubkey
+    )
+
+    # Now Bob can retrieve the original message.
+    alice_pubkey_restored_from_ancient_scroll = UmbralPublicKey.from_bytes(alices_pubkey_bytes_saved_for_posterity)
+    delivered_cleartexts = BOB.retrieve(single_passage_ciphertext,
+                                        enrico=enrico_as_understood_by_bob,
+                                        alice_verifying_key=alice_pubkey_restored_from_ancient_scroll,
```

Comment:
There's a strange term transformation going on. First, we have "Alice's stamp", which we save as "Alice's pubkey" and later use as "Alice's verifying key". Could they all be named the same?

Reply:
:heavy_check_mark: